scripts/Tools/pointclm/pointCLM.py

```diff
@@ -132,36 +132,23 @@
 #                   default=False,action="store_true")
 parser.add_option("--nofire", action="store_true", dest="nofire", default=False, \
                     help="To turn off wildfires")
-parser.add_option("--npoolmod", action="store_true", dest="npoolmod", default=False, \
-                    help="To turn on nitrogen pool modifications")
-parser.add_option("--cpoolmod", action="store_true", dest="cpoolmod", default=False, \
-                    help="To turn on carbon storage pool modifications")
-
-parser.add_option("--q10wbmod", action="store_true", dest="q10wbmod", default=False, \
```

Comment:
​OK, there are several different flavors of single-point CLM. It sounds
like this is referring to the Python version of PTCLM that created CLM
datasets, and then setup a case for it. We decided that having a tool that
does both data-set creation AND interacts with cime/scripts to setup a case
-- was a bad idea.

So we split the functionality and we now have a Python script under:

components/clm/tools/shared/PTCLM

called PTCLMmkdata

that does the dataset creation part. It both creates the datasets as well
as creates user_nl_clm and shell_commands files that can be used in
conjunction with "create_newcase" in scripts. This is a much more
sustainable way to make the same functionality work. When it was hardwired
to "create_newcase" it seemed like it broke all the time. And the step of
creating data is a one-time step, while creating cases from data you've
setup is something you do a lot of. So it just made sense to separate the
two.

The ACME version of CLM and CIME is going to have both the PTCLM.py script
as well as PTCLMmkdata. Do you really want both? And do you really want to
have a script that is hardwired into it's use of "create_newcase"? We
decided that was a bad plan and went away from it -- does ACME really want
to retain that? Totally just questions for ACME to figure out. I'm just
giving the history and pointing out that you have both, and why we went to
the PTCLMmkdata method. And I am suggesting you might want to make the same
shift as well.

Thanks.

Feel free to ask me questions about all of this...

​

*Erik Kluzek, (CGD at NCAR)*

*National Center for Atmospheric Research*

*Boulder CO, *

*(off) (303)497-1326 (fax) (303)497-1348*

*(skype) ekluzek    (cell) (303)859-0183*

*Pronouns: he/his/him*

*------------------ Home page ------------------------*

*          http://www.cgd.ucar.edu/~erik <http://www.cgd.ucar.edu/~erik>*

*          https://staff.ucar.edu/users/erik
<https://staff.ucar.edu/users/erik>*

*!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!*

On Wed, May 4, 2016 at 4:17 PM, James Foucar <notifications@github.com>
wrote:

> In scripts/Tools/pointclm/pointCLM.py
> <https://github.com/ESMCI/cime/pull/155#discussion_r62124399>:
>
> > @@ -132,36 +132,23 @@
> >  #                   default=False,action="store_true")
> >  parser.add_option("--nofire", action="store_true", dest="nofire", default=False, \
> >                      help="To turn off wildfires")
> > -parser.add_option("--npoolmod", action="store_true", dest="npoolmod", default=False, \
> > -                    help="To turn on nitrogen pool modifications")
> > -parser.add_option("--cpoolmod", action="store_true", dest="cpoolmod", default=False, \
> > -                    help="To turn on carbon storage pool modifications")
> > -
> > -parser.add_option("--q10wbmod", action="store_true", dest="q10wbmod", default=False, \
>
> @mvertens <https://github.com/mvertens> , it is OK if we move after ACME
> adopts ESMCI?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ESMCI/cime/pull/155/files/8755d2cf6839980bded285732c57401465227a80#r62124399>
>